src/SapNwRfc/ISapFunction.cs

```diff
@@ -14,6 +14,12 @@ public interface ISapFunction : IDisposable
         /// <returns>Whether the parameter name exists.</returns>
         bool HasParameter(string parameterName);
 
+        /// <summary>
+        /// Gets the metadata for this function.
+        /// </summary>
+        /// <returns>The matadata for this function.</returns>
+        ISapFunctionMetadata GetMetadata();
```

Comment:
I'd just make this a property called `Metadata` backed by a `Lazy<ISapFunctionMetadata>`, so the low-level SAP function is only called on the first property 'get'.